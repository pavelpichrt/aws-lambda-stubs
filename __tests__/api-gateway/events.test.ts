import {
  apiGatewayProxyEvent,
  apiGatewayAuthorizerEvent,
} from '../../src/api-gateway/events';

const nonEmptyAlNumStringRegex = /[a-zA-Z0-9]+/;

describe('apiGatewayProxyEvent', () => {
  it('Generates a correct stub', () => {
    const { headers, requestContext, multiValueHeaders, ...baseParams } =
      apiGatewayProxyEvent();
    const {
      Authorization,
      'X-Amzn-Trace-Id': traceId,
      deterministicHeaders,
    } = headers;
    const {
      requestId,
      requestTime,
      requestTimeEpoch,
      resourceId,
      extendedRequestId,
      authorizer,
      ...deterministicContext
    } = requestContext;

    expect(baseParams).toMatchSnapshot();
    expect(deterministicHeaders).toMatchSnapshot();
    expect(deterministicContext).toMatchSnapshot();
    expect(multiValueHeaders['Content-Type']).toMatchSnapshot();
    expect(Authorization).toEqual(
      expect.stringMatching(nonEmptyAlNumStringRegex),
    );
    expect(traceId).toEqual(expect.stringMatching(nonEmptyAlNumStringRegex));
    expect(requestId).toEqual(
      expect.stringMatching(
        /^[0-9(a-f|A-F)]{8}-[0-9(a-f|A-F)]{4}-4[0-9(a-f|A-F)]{3}-[89ab][0-9(a-f|A-F)]{3}-[0-9(a-f|A-F)]{12}$/,
      ),
    );
    expect(requestTime).toEqual(
      expect.stringMatching(nonEmptyAlNumStringRegex),
    );
    expect(requestTimeEpoch).toEqual(expect.any(Number));
    expect(resourceId).toEqual(expect.stringMatching(/[a-z0-9]{6}$/));
    expect(extendedRequestId).toEqual(
      expect.stringMatching(nonEmptyAlNumStringRegex),
    );
    expect(authorizer?.integrationLatency).toEqual(expect.any(Number));
  });
});

describe('apiGatewayAuthorizerEvent', () => {
  it('Generates a correct stub', () => {
    const actualEvent = apiGatewayAuthorizerEvent();

    expect(actualEvent).toMatchSnapshot();
  });
});
