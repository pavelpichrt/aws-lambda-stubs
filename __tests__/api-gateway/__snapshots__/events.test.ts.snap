// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`apiGatewayAuthorizerEvent Generates a correct stub 1`] = `
{
  "authorizationToken": "authorization-token-stub",
  "methodArn": "arn:aws:execute-api:eu-west-1:123456789012:example/prod/POST/{proxy+}",
  "type": "TOKEN",
}
`;

exports[`apiGatewayProxyEvent Generates a correct stub 1`] = `
{
  "body": null,
  "httpMethod": "POST",
  "isBase64Encoded": false,
  "multiValueQueryStringParameters": null,
  "path": "/mock-stage//mock/foo/path",
  "pathParameters": {
    "id": "foo",
  },
  "queryStringParameters": null,
  "resource": "/mock/{id}/path",
  "stageVariables": null,
}
`;

exports[`apiGatewayProxyEvent Generates a correct stub 2`] = `undefined`;

exports[`apiGatewayProxyEvent Generates a correct stub 3`] = `
{
  "accountId": "123456278901",
  "apiId": "mockapi",
  "domainName": "mockapi.execute-api.eu-west-2.amazonaws.com",
  "domainPrefix": "mockapi",
  "httpMethod": "POST",
  "identity": {
    "accessKey": null,
    "accountId": null,
    "apiKey": null,
    "apiKeyId": null,
    "caller": null,
    "clientCert": null,
    "cognitoAuthenticationProvider": null,
    "cognitoAuthenticationType": null,
    "cognitoIdentityId": null,
    "cognitoIdentityPoolId": null,
    "principalOrgId": null,
    "sourceIp": "123.123.123.123",
    "user": null,
    "userAgent": "MockApi/1.0.0",
    "userArn": null,
  },
  "path": "/mock-stage//mock/foo/path",
  "protocol": "HTTP/1.1",
  "resourcePath": "/mock/{id}/path",
  "stage": "mock-stage",
}
`;

exports[`apiGatewayProxyEvent Generates a correct stub 4`] = `
[
  "application/json",
]
`;
